File: leebroscode.com/보고 말하는 수열.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
	int n;cin>>n;
	
	string num="1";
	
	while(n--){
		string temp;
		stack <char> S;
		int count=0;
		for(int i=0;i<num.length();i++){
			char number=num[i];
			cout<<number<<"\n";
			if(S.empty()){
				S.push(number);
				count++;
			}else{
				if(S.top()==number){//젤 위에 있는 수와 같으면. 
					count++;
				}else{//젤 위에 있는 수와 다르면. 
					temp.push_back(S.top());
					char num=count;
					temp.push_back(num);
					S.pop();
					S.push(number);
					count=1;
				}
			}
		}
		num=temp;
	}
	cout<<num;
}

File: leebroscode.com/보고 말하는 수열.py
n=int(input())

num='1';

dict={}
if(n==1): print('1')
else:
    n-=1
    while(n>0):
        ans='';
        for i in range(1, 10): dict[i] = 0;
        for i in range(len(num)):
            number=int(num[i])
            if(dict[number]>0):#저장된 값과 동일한 경우.
                dict[number]+=1
            else:#저장된 값과 다른 경우.
                for j in range(1,10):
                    if(dict[j]>0):
                        ans+=str(j);ans+=str(dict[j]);
                        dict[j]=0
                        break
                dict[number] = 1
        for j in range(1, 10):
            if (dict[j] > 0):
                ans += str(j);
                ans += str(dict[j]);
        num=ans
        n-=1

    print(ans)